File: js-python-challenges/longest_common_prefix.py
"""Write a function to find the longest common prefix string amongst an array of strings."""


def longest_common_prefix(strings: list[str]) -> str:
    if not strings:
        return ""

    prefix = strings[0]
    for string in strings:
        while string.find(prefix, 0) != 0:
            prefix = prefix[:-1]
            if len(string) < 1:
                return ""
    return prefix


if __name__ == "__main__":
    strings = ["flower", "flow", "flight"]
    result = longest_common_prefix(strings)
    print(result)

File: js-python-leetcode-challenges/longestCommonPrefix.js
/**
 * Write a function to find the longest common prefix string amongst an array of strings.
 *
 * @param {string[]} strs
 * @returns {string}
 */

function longestCommonPrefix(strs) {
  if (strs.length < 1) return ""
  let prefix = strs[0]
  for (let i = 1; i < strs.length; i++) {
    while (strs[i].indexOf(prefix, 0) !== 0){
      prefix = prefix.slice(0, -1)
      if (prefix < 1) return ""
    }
  }
  return prefix
}

const strs = ["flower","flow","flight"]
console.log(longestCommonPrefix(strs))
